### ISSUE TITLE
Qwen2.5-Omni Token2Wav rebuilds state_dict once per buffer

### ISSUE THREAD

**Issue by akshatvishu** [vllm-project/vllm-omni#7060]
## Environment

| Item | Value |
| --- | --- |
| Model | `Qwen/Qwen2.5-Omni-3B` |
| GPU | 1 AMD Instinct MI300X VF |
| PyTorch | `2.12.0+git6bbd260` |
| ROCm | `7.2.53211` |
| vLLM | `0.28.0+rocm723` |
| vLLM-Omni | `0.28.1.dev47` |

All three stages used device 0 with GPU memory utilization values of 0.35, 0.25, and 0.15. The model files were already cached.

## Issue

`Qwen2_5OmniToken2WavForConditionalGenerationVLLM.find_all_registers()` calls `state_dict()` inside the loop over `named_buffers()`:

```python
for name, buf in self.named_buffers():
    if name in self.state_dict():
        registers[name] = {"name": name, "buffer": buf}
```

The membership check has a valid purpose. `named_buffers()` includes persistent and nonpersistent buffers, while `state_dict()` includes only persistent buffers. Calling `state_dict()` for every buffer rebuilds the full state dictionary and repeats the same module traversal.

Qwen2.5-Omni made 219 `state_dict()` calls during each measured startup.

## Proposed change

Collect the fully qualified names of nonpersistent buffers once, then walk `named_buffers()` once:

```python
non_persistent_names = set()
for module_name, module in self.named_modules():
    for buffer_name in getattr(module, "_non_persistent_buffers_set", set()):
        name = f"{module_name}.{buffer_name}" if module_name else buffer_name
        non_persistent_names.add(name)

for name, buffer in self.named_buffers():
    if name not in non_persistent_names:
        registers[name] = {"name": name, "buffer": buffer}
```

The change preserves the original persistent buffer behavior and removes all 219 calls to `state_dict()`.

## Measurements

The benchmark ran one warmup per version and then five measured runs per version. Runs alternated between the baseline and optimized commits. Warmups were excluded from the medians.

| Measurement | Baseline median | Optimized median | Change |
| --- | ---: | ---: | ---: |
| Buffer discovery | 456.946 ms | 1.564 ms | 455.382 ms faster, 99.66% |
| Token2Wav weight loading | 1099.526 ms | 620.436 ms | 479.090 ms faster, 43.57% |
| Startup to `/health` | 105.738 s | 104.322 s | 1.416 s faster, 1.34% |

The startup ranges overlapped, from 104.214 to 106.067 seconds for the baseline and from 103.660 to 105.981 seconds after the change. The direct Token2Wav measurements show the improvement. The full startup measurement includes about 104 seconds of other work and does not isolate the change from normal run variation.

## Why this change

vLLM already checks `_non_persistent_buffers_set` when it collects persistent registered buffers. vLLM-Omni uses the same check in diffusion weight loaders. Applying the existing pattern here avoids a new abstraction and keeps the same buffer selection behavior.
